exec/java-exec/src/main/java/org/apache/drill/exec/store/ColumnExplorer.java

```diff
@@ -224,6 +224,33 @@ public static boolean isPartitionColumn(String partitionDesignator, String path)
     return Arrays.asList(diffDirectoryNames);
   }
 
+  public static List<String> listPartitionValuesDirectory(String dirPath, String root) {
```

Comment:
It copies 90% of `listPartitionValues()` method. Could you just pass some boolean flag to the old method, for example   `hasDirsOnly` as [here](https://github.com/apache/drill/blob/master/exec/java-exec/src/main/java/org/apache/drill/exec/planner/DFSFilePartitionLocation.java#L29)?